client/network/src/discovery.rs

```diff
@@ -123,6 +126,12 @@ impl DiscoveryConfig {
 		self
 	}
 
+	/// Should non-global addresses be inserted to the DHT?
+	pub fn allow_non_globals_in_dht(&mut self, value: bool) -> &mut Self {
+		self.allow_non_globals_in_dht = value;
+		self
+	}
```

Comment:
Given that it defaults to false, why not:

```suggestion
	pub fn allow_non_globals_in_dht(&mut self) -> &mut Self {
		self.allow_non_globals_in_dht = true;
		self
	}
```

Reply:
The current implementation is consistent with `allow_private_ipv4`, so we should then change both. I do not think it makes much of a difference so unless you (or anyone else) feels strongly about it I would probably leave it as is.